docs/source/jobs.rst

```diff
@@ -239,41 +239,127 @@ Use cases for the **job document** include, but are not limited to:
 
 Job Data Storage
 ================
-
-Basics
-------
-
-Large numerical or text data can be stored in the :py:attr:`Job.data` container, which is an instance of :class:`signac.H5Store`.
-This container uses a file in `HDF5`_ format to store array-like or dictionary-like information.
+Large numerical or text data can be stored in the :py:attr:`Job.data` container, which is an instance of :class:`signac.H5Store`. This container uses a file in `HDF5`_ format to store array-like or dictionary-like information.
 Like the :py:attr:`Job.document`, this information can be accessed using key-value pairs.
 Unlike the :py:attr:`Job.document`, :attr:`Job.data` is not searchable.
-
 .. _`HDF5`: https://portal.hdfgroup.org/display/HDF5/HDF5
 
+Data written with :py:attr:`Job.data` is stored in a file named `signac_data.h5` in the associated `job` folder. For cases where `job`-associated data may be accessed from multiple sources at the same time or other instances where multiple files may be preferred to one large file, :py:attr:`Job.stores` should be used instead of :py:attr:`Job.data`.
+
+Reading and Writing data
+-----------------------
+
 An example of storing data:
 
 .. code-block:: python
 
     >>> import numpy as np
     >>> job = project.open_job(statepoint)
-    >>> job.data['x'] = np.arange(0, 1, 0.01)
-    >>> job.data['hello'] = 'world'
+    >>> job.data['x'] = np.ones([10, 3, 4])
+
 
 Just like the job *state point* and *document*, individual keys may be accessed either as attributes or through a functional interface, *e.g.*.
-The following examples are all equivalent:
+
+To access data as attributes:
 
 .. code-block:: python
 
-    >>> print(job.data.get('hello'))
-    world
-    >>> print(job.data['hello'])
-    world
-    >>> print(job.data.hello)
-    world
+    >>> with job.data:
+    ...     x = job.data.x[:]
+
+To access data as a key:
+
+.. code-block:: python
+
+    >>> with job.data:
+    ...     x = job.data['x'][:]
+
+Through a functional interface:
+
+.. code-block:: python
+    
+    >>> with job.data:
+    ...     x = job.data.get('x')[:]
 
 .. tip::
 
-     Use the :py:meth:`Job.data.get` method to return ``None`` or another specified default value for missing values. This works exactly like with python's `built-in dictionaries <https://docs.python.org/3/library/stdtypes.html#dict.get>`_.
+     Use the :py:meth:`Job.data.get` method to return ``None`` or another specified default value for missing values. This works exactly like with python's `built-in dictionaries <https://docs.python.org/3/library/stdtypes.html#dict.get>`_. 
```

Comment:
Please also make this change in the job document docs (further up in this file) and any other place it appears. We shouldn't hard-code links to the Python docs since we have intersphinx configured.
```suggestion
     Use the :py:meth:`Job.data.get` method to return ``None`` or another specified default value for missing values. This works exactly like with Python's built-in dictionaries (see :py:meth:`dict.get`). 
```